README.rst

```diff
@@ -72,3 +67,219 @@ response, e.g.
 .. code:: python
 
    WAGTAIL_STORAGES_DOCUMENT_HOOK_ORDER = 900
+
+
+Recommended S3 setup with Wagtail
+---------------------------------
+
+This is a guide that explains the recommended setup for using S3 with Wagtail.
+This guide assumes that:
+
+* You serve your main website at ``llamasavers.com``.
+* Your S3 bucket is called ``media.llamasavers.com`` and you host it from that
+  domain name.
+* Using CDN on that domain, this guide will assume Cloudflare.
+
+Set up S3 bucket
+~~~~~~~~~~~~~~~~
+
+First step is to set up your S3 bucket. It must be configured to:
+
+* Bucket name must match the domain name, e.g. ``media.llamasavers.com``, that
+  you intend using it with.
+* Allow user to perform the following actions on the bucket:
+  * ``s3:ListBucket``
+  * ``s3:GetBucketLocation``
+  * ``s3:ListBucketMultipartUploads``
+  * ``s3:ListBucketVersions``
+* Allow user to perform all the actions (``s3:*``) on the objects within the
+  bucket.
+* Allow the internet traffic to access Wagtail image renditions (``images/*``).
+
+The user permissions can be set in the IAM or via a bucket policy. See example
+of all of those points being achieved with a bucket policy below.
```

Comment:
"See example of all of those points being achieved with a bucket policy below." → "The following example shows all the above being achieved with a bucket policy:"